### DISCUSSION TITLE
[OP Stack Deployment] How does Op-challenger work

### DISCUSSION THREAD

**Discussion by DanielW8088** [ethereum-optimism/developers#392] [Deploying the OP Stack] [answered]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I started an op stack chain of version 1.7.2
When I withdraw, I need proof of fraud. There are no documents here on how to send game messages.
I try to run this command, it seems there is no response.
What should I do to start the game?

./op-challenger/bin/op-challenger \
  --trace-type cannon \
  --l1-eth-rpc <> \
  --l1-beacon <> \
  --rollup-rpc http://localhost:8547 \
  --game-factory-address <> \
  --datadir temp/challenger-data \
  --cannon-rollup-config rollup.json  \
  --cannon-l2-genesis genesis-l2.json \
  --cannon-bin ./cannon/bin/cannon \
  --cannon-server ./op-program/bin/op-program \
  --cannon-prestate ./op-program/bin/prestate.json \
  --cannon-l2 http://localhost:8545 \
  --private-key <> \
  --num-confirmations 1
  
  

### Logs

challenger logs
INFO [05-08|23:42:15.060] Starting op-challenger                   version=v0.1.0-99a53381-1710933867-dev
INFO [05-08|23:42:15.208] starting scheduler
INFO [05-08|23:42:15.208] starting monitoring
INFO [05-08|23:42:15.208] challenger game service start completed

  ./bin/op-challenger create-game \
  --l1-eth-rpc <> \
  --game-factory-address <> \
  --output-root <> \
  --l2-block-num<> \
  --private-key <> 
  WARN [05-08|12:23:27.014] Failed to create a transaction, will retry service=challenger err="failed to estimate gas: execution reverted"
WARN [05-08|12:23:30.150] Failed to create a transaction, will retry service=challenger err="failed to estimate gas: execution reverted"

**Comment by DanielW8088** [comment]
proposer command, do I need to adjust anything?
op-proposer \
    --poll-interval=12s \
    --rpc.port=8560 \
    --rollup-rpc=http://localhost:8547 \
    --l2oo-address=$L2OO_ADDRESS \
    --private-key=$PROPOSER_KEY \
    --l1-eth-rpc=$L1_ETH_RPC_URL

**Reply by Chomtana** [reply]
I think it shouldn't use L2OO (L2 Output Oracle) anymore

Try replacing `--l2oo-address` with `--game-factory-address`

**Reply by DanielW8088** [reply]
I tried this, but it didn't work.
INFO [05-10|03:47:21.613] Proposer started
WARN [05-10|03:47:45.630] Failed to create a transaction, will retry service=proposer err="failed to estimate gas: execution reverted"
WARN [05-10|03:47:47.636] Failed to create a transaction, will retry service=proposer err="failed to estimate gas: execution reverted"
WARN [05-10|03:47:49.645] Failed to create a transaction, will retry service=proposer err="failed to estimate gas: execution reverted"
WARN [05-10|03:47:51.649] Failed to create a transaction, will retry service=proposer err="failed to estimate gas: execution reverted"
WARN [05-10|03:47:53.655] Failed to create a transaction, will retry service=proposer err="failed to estimate gas: execution reverted"

**Reply by Chomtana** [reply]
That's weird; can you try setting the log level to trace for more verbosity, and hope it will reveal the revert reason or transaction body that we can input into Tenderly for simulation and debugging?

`--log.level=trace` on op-proposer

Tagging @sbvegan here. The challenger is much more advanced than my knowledge is!

**Reply by DanielW8088** [reply]
WARN [05-14|02:22:52.021] Failed to create a transaction, will retry service=proposer err="failed to estimate gas: execution reverted"
DEBUG[05-14|02:22:54.022] crafting Transaction                     service=proposer blobs=0 calldata_size=164
WARN [05-14|02:22:54.026] Failed to create a transaction, will retry service=proposer err="failed to estimate gas: execution reverted"
DEBUG[05-14|02:22:56.026] crafting Transaction                     service=proposer blobs=0 calldata_size=164
WARN [05-14|02:22:56.031] Failed to create a transaction, will retry service=proposer err="failed to estimate gas: execution reverted"

After research the source code, I found that the contract call failed.
contract name: game-factory-address, the request parameters of this contract are as follows
Function: create(uint32 _gameType,bytes32 _rootClaim,bytes _extraData) ***
Inputdata
0 | _gameType | uint32 | 1
1 | _rootClaim | bytes32 | 0x60b975b71885368b28b6d1978877d21c4eee65a8c59b93af433af12eda541a57
2 | _extraData | bytes | 0x00000000000000000000000000000000000000000000000000000000000dd131
Is there anything wrong?

**Reply by DanielW8088** [reply]
@sbvegan  Could you provide some help, please?

**Reply by Chomtana** [reply]
_gameType looks weird. OP sepolia use gameType 0

https://sepolia.etherscan.io/tx/0x50112228bb5198837469667265735d3635f53fd4d393651cea8d3eeb32e45e6b

**Reply by DanielW8088** [reply]
I tried 0, failed the same way

**Reply by DanielW8088** [reply]
0 | _gameType | uint32 | 0
1 | _rootClaim | bytes32 | 0x60b975b71885368b28b6d1978877d21c4eee65a8c59b93af433af12eda541a57
2 | _extraData | bytes | 0x00000000000000000000000000000000000000000000000000000000000dd131


Warning! Error encountered during contract execution [execution reverted]

**Reply by Chomtana** [reply]
Or you may try on [tenderly](https://tenderly.co/) as it can breakdown which line causes the revert?

**Comment by joohhnnn** [comment, marked as answer]
Hi there you can refer to this https://github.com/joohhnnn/The-book-of-optimism-fault-proof/blob/main/05-op-challenger.md

**Comment by opfocus** [comment]
If there is already an answer you find acceptable, you can mark it as the accepted answer (you can also mark your own answer as accepted), and then close the discussion. You can also close it without marking any answer. Additionally, you can continue to add comments.
